algorithms/src/Kokkos_Sort.hpp

```diff
@@ -298,6 +299,18 @@ class BinSort {
                     values.extent(5),
                     values.extent(6),
                     values.extent(7));
+#else
+    scratch_view_type
+      sorted_values("Scratch",
+                  values.rank_dynamic > 0 ? len : KOKKOS_CTOR_DEFAULT_ARG,
+                  values.rank_dynamic > 1 ? values.extent(1) : KOKKOS_CTOR_DEFAULT_ARG ,
```

Comment:
Lets rename KOKKOS_CTOR_DEFAULT_ARG to KOKKOS_IMPL_CTOR_DEFAULT_ARG since we want it to go away again and be replaced by KOKKOS_INVALID_INDEX as soon as as we eliminate deprecated code

Reply:
Done.